lib/plugins/index.js

```diff
@@ -216,7 +216,15 @@ function init (ctx) {
   plugins.extendedClientSettings = function extendedClientSettings (allExtendedSettings) {
     var clientSettings = {};
     _each(clientDefaultPlugins, function eachClientPlugin (plugin) {
-      clientSettings[plugin.name] = allExtendedSettings[plugin.name];
+      var settings = allExtendedSettings[plugin.name];
+      if(settings) {
+        // sanitize private settings
+        var settingscopy = JSON.parse(JSON.stringify(settings));
+        delete settingscopy.apnsKey;
```

Comment:
Rather than hardcoding the settings here, we should extend on filteredSettings() in settings.js and probably implement the filtering in settings.js, which is a better domain for filtering than the plugins.

Reply:
Actually looks like the only thing that's needed here is adding passing allExtendedSettings through settings.filteredSettings() in the beginning of the function as the rest of this is already implemented there